cirrus/google_cloud/manager.py

```diff
@@ -44,6 +44,21 @@
     'STANDARD'  # alias for MULTI_REGIONAL/REGIONAL, based on location
 ]
 
+COMPUTE_ENGINE_DEFAULT_SERVICE_ACCOUNT = (
+    'COMPUTE_ENGINE_DEFAULT_SERVICE_ACCOUNT'
+)
+GOOGLE_API_SERVICE_ACCOUNT = 'GOOGLE_API_SERVICE_ACCOUNT'
+COMPUTE_ENGINE_API_SERVICE_ACCOUNT = 'COMPUTE_ENGINE_API_SERVICE_ACCOUNT'
+USER_MANAGED_SERVICE_ACCOUNT = 'USER_MANAGED_SERVICE_ACCOUNT'
+
+GOOGLE_SERVICE_ACCOUNT_DOMAIN_TYPE_MAPPING = [
+    ('appspot.gserviceaccount.com', COMPUTE_ENGINE_DEFAULT_SERVICE_ACCOUNT),
+    ('cloudservices.gserviceaccount.com', GOOGLE_API_SERVICE_ACCOUNT),
+    ('compute-system.iam.gserviceaccount.com',
+     COMPUTE_ENGINE_API_SERVICE_ACCOUNT),
+    ('iam.gserviceaccount.com', USER_MANAGED_SERVICE_ACCOUNT),
```

Comment:
this is a little scary since order is important here in regards to whether or not it's compute engine or user managed. I'd add a comment for future devs to NOT change the order of these items since logic depends on it just to be safe

Reply:
Added, just not registered by github since it's above the highlighted section :smile: